addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -252,21 +258,11 @@ function factory(dependencies) {
          */
         _computeIsViewable() {
             return (
-                this.mediaType === 'image' ||
-                this.mediaType === 'video' ||
-                this.mimetype === 'application/pdf' ||
-                this.isTextFile
+                ['image', 'text', 'video'].includes(this.fileType) ||
```

Comment:
In theory recognized file types being an image/text/video should not simply be enough for stating it's viewable.

I have some issues with filetype as is, and would gladly just drop it altogether.

Code that relied on `fileType` should use something else, e.g. explicit list of supported mimetypes or introduce boolean fields that on files that represent a well-defined class of files. For instance, `isTextFile` should rely on list of mimetypes that are actually considered as text file in our code. It doesn't necessarily means all text files are viewable. I think `docx` is considered a text file but it shouldn't be viewable.

TBH the notion of "is text" file is not well defined enough, so I would prefer to drop it if possible. For instance, it's strange that a PDF is never considered a text file, when I'm pretty sure the majority of person would consider most PDF files being text files.

Reply:
I agree that the "type" of the file does not mean it is viewable, both should be separated.

It's actually dangerous to change `fileType` in this PR because it is used for other purpose than just deciding if it's viewable or not.
Eg. `_computeDefaultSource` is heavily reliant on this type, and I'm pretty sure the current PR will break it.

Reply:
Let's keep for the time being then.